### ISSUE TITLE
Number of iterations saturating at 29.

### ISSUE THREAD

**Issue by sushmit-goyal** [microsoft/LightGBM#5275]
I'm trying to continue training the booster model by passing saved model in `init_model` in the train function. I am observing that no new trees are added after 29 trees. Even if I call the train function over and over again. What could be the issue? The parameters I'm using are:
`'seed': 0,
    'num_leaves': 12,
    'learning_rate': 1,
    'is_unbalance': 'true',
    'boosting_type' : 'gbdt',
    'application' : 'binary',
    'feature_fraction' : 1,
    'bagging_fraction' : 1,
    'n_estimators': 10,
    'min_split_gain' : 1,
    'min_data_in_leaf': 100,
    'max_depth' : 7,
    'verbose' : -1,`
without any early stopping.

**Comment by jmoralez** [comment]
Hi @sushmit-goyal, thank you for your interest in LightGBM. Are you able to provide a [minimal reproducible example](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports)? Otherwise we're only able to guess. So my guess is that the `min_split_gain = 1` condition isn't being satisfied so no new trees are being added but it could be something else.
